pkg/stub/handler.go

```diff
@@ -152,88 +155,98 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 	case *corev1.Secret:
 		dockercfgSecret, _ := event.Object.(*corev1.Secret)
 
-		if h.operatorCfg == nil {
+		if h.samplesResource == nil {
 			dockercfgSecret.DeepCopyInto(&h.registrySecret)
```

Comment:
since we don't know the secretname that's going to be referenced by the sampleresource (because sampleresource is nil), how do we know we actually care about this secret?  (there could be multiple secrets in the namespace).

This is part of where i was going w/ wondering if us hardcoding the secret name would make your life easier.

Alternatively, if we want to keep the secretname as part of the resource spec, maybe it would make more sense to implement the logic as:

1) if we see a secret event, and we don't know the name of the secret we care about yet (because we haven't seen the samples resource yet) do nothing.
2) if we see a secret event we have seen the resource and this secret is the one we want, process it
3) if we see the sampleresource object and we *haven't* processed/found the secret yet, check to see if the secret referenced by the samplesresource exists.  If so, retrieve it and process it immediately (because clearly we already missed/ignored the creation event and we don't want to wait for a relist).

Reply:
I'm going to switch to having the secret name be a constant